pkg/theiactl/commands/check.go

```diff
@@ -72,12 +78,88 @@ $ theiactl policyreco check --id e998433e-accb-4888-9fc8-06563f073e86
 			return err
 		}
 		state := strings.TrimSpace(string(sparkApplication.Status.AppState.State))
+		if state == "RUNNING" {
+			// Check the working progress of running recommendation job
+			// Forward the policy recommendation service port
+			portForwardCmd := exec.Command("kubectl", "port-forward", fmt.Sprintf("service/policy-reco-%s-ui-svc", recoID), "-n", flowVisibilityNS, "4040:4040")
+			if err := portForwardCmd.Start(); err != nil {
+				return fmt.Errorf("fail to forward port for policy recommendation service, %v", err)
+			}
+			defer portForwardCmd.Process.Kill()
+			stateProgress, err := getPolicyRecommendationProgress(recoID)
+			if err != nil {
+				return err
+			}
+			state += stateProgress
+		}
 		fmt.Printf("Status of this policy recommendation job is %s\n", state)
 		return nil
-		// TODO: add implementation of checking work progress through Spark Monitoring Service after port forwarder finished
 	},
 }
 
+func getPolicyRecommendationProgress(id string) (string, error) {
+	// Get the id of current spark application
+	url := "http://localhost:4040/api/v1/applications"
+	response, err := getResponseFromSparkMonitoringSvc(url)
+	if err != nil {
+		return "", fmt.Errorf("failed to get response from Spark Monitoring service with id %s, %v", id, err)
+	}
+	var getAppsResult []map[string]interface{}
+	json.Unmarshal([]byte(response), &getAppsResult)
+	if len(getAppsResult) != 1 {
+		return "", fmt.Errorf("wrong number of Spark Application with id %s, expected 1, got %d", id, len(getAppsResult))
+	}
+	sparkAppID := getAppsResult[0]["id"]
+	// Check the percentage of completed stages
+	url = fmt.Sprintf("http://localhost:4040/api/v1/applications/%s/stages", sparkAppID)
+	response, err = getResponseFromSparkMonitoringSvc(url)
+	if err != nil {
+		return "", fmt.Errorf("failed to get response from Spark Monitoring service at %s, %v", url, err)
+	}
+	var getStagesResult []map[string]interface{}
+	json.Unmarshal([]byte(response), &getStagesResult)
+	if len(getStagesResult) < 1 {
+		return "", fmt.Errorf("wrong number of Spark Application stages, expected more than 1, got %d", len(getStagesResult))
+	}
+	completedStages := 0
+	for _, stage := range getStagesResult {
+		if stage["status"] == "COMPLETE" || stage["status"] == "SKIPPED" {
+			completedStages++
+		}
+	}
+	return fmt.Sprintf(": %d/%d (%d%%) stages completed", completedStages, len(getStagesResult), completedStages*100/len(getStagesResult)), nil
```

Comment:
pedant comment: calculate `len(getStageResult)` once and then use value in the return statement above